src/Bicep.Core/TypeSystem/TypeTypeProperty.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Bicep.Core.TypeSystem
+{
+    public class TypeTypeProperty : TypeProperty
```

Comment:
I am not good at naming things 😞. This class exists so that `NamespaceType` can enforce that all of its properties are themselves types, which is represented as a `TypeType` (another name that could probably use some work).

Reply:
I don't mind the names, but would be good to add comments on the classes to describe what they represent.